### PR TITLE
Add documentation about ions.

### PR DESCRIPTION
**Pull request by 1uc**
CI_BRANCHES:NEURON_BRANCH=pramodk/random-test-fix

### CODE REVIEW

File: docs/contents/ions.rst [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    ina = gna*(v - ena)
+  }
+
+Semantically, this states that the contribution of the Hodgkin-Huxley model to
+the overall sodium current in that segment is ``gna*(v - ena)``. Since
+everything is at the level of segment, we'll not keep repeating "for that
+segment". Similarly, each mechanism computes a `local` contribution, i.e. the
+contribution due to this mechanism to the actual `global` ion current.
+
+Therefore, the following code for the following must be generated:
```
Comment (1uc) [inline docs/contents/ions.rst:63]:
```suggestion
Therefore, code for the following must be generated:
```

File: docs/contents/ions.rst [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+1. Compute the local contribution to the sodium current.
+2. Increment the total, local, current contribution.
```
Comment (1uc) [inline docs/contents/ions.rst:66]:
```suggestion
2. Increment the total, local, current contribution by ``ina``.
```

File: docs/contents/ions.rst [hunk 3]
```diff
@@ -0,0 +1,109 @@
+Ions
+====
+
+NEURON supports computing the ion currents. For each segment there can be
```
Comment (nrnhines) [inline docs/contents/ions.rst:4]: ...computing the ion currents and ion concentrations.

File: docs/contents/ions.rst [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+separate current for separate ions, i.e. one for sodium ions and another for
+calcium ions.
+
+There are five variables associated with ions: the current (``ina``), the
+concentration inside the segment (``nai``), the concentration outside the
```
Comment (nrnhines) [inline docs/contents/ions.rst:9]: ... inside the segment adjacent to the membrane

... outside the segment adjacent to the membrane

(the reason I mention it is that many ion accumulation models discretize the radial dimension to model radial diffusion.

**Review by nrnhines — APPROVED** [review]
Good job. My edit suggestions are not really needed for this document.

File: docs/contents/ions.rst (hunk 3 above)
Comment (1uc) [reply docs/contents/ions.rst]: added.

File: docs/contents/ions.rst (hunk 4 above)
Comment (1uc) [reply docs/contents/ions.rst]: Added, thank you for the useful explanation.

**Comment by 1uc** [comment]
Thanks to everyone for proof reading.

**Review by pramodk — APPROVED** [review]
🙌

**Comment by pramodk** [comment]
Note: just have retrigger CI once https://github.com/neuronsimulator/nrn/pull/2717 is merged.

**Pull request merged**